torchmetrics/aggregation.py

```diff
@@ -0,0 +1,444 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from typing import Any, Callable, List, Optional, Union
+
+import torch
+from torch import Tensor
+
+from torchmetrics.metric import Metric
+from torchmetrics.utilities.data import dim_zero_cat
+
+
+class BaseAggregator(Metric):
+    """Base class for aggregation metrics.
+
+    Args:
+        fn: string specifying the reduction function
+        default_value: default tensor value to use for the metric state
+        nan_strategy:
+            - ``'error'``: if any `nan` values are encounted will give a RuntimeError
+            - ``'warn'``: if any `nan` values are encounted will give a warning and continue
+            - ``'ignore'``: all `nan` values are silently removed
+            - a float: if a float is provided will impude any `nan` values with this value
+
+        compute_on_step:
+            Forward only calls ``update()`` and returns None if this is
+            set to False. default: True
+        dist_sync_on_step:
+            Synchronize metric state across processes at each ``forward()``
+            before returning the value at the step.
+        process_group:
+            Specify the process group on which synchronization is called.
+            default: None (which selects the entire world)
+        dist_sync_fn:
+            Callback that performs the allgather operation on the metric state.
+            When `None`, DDP will be used to perform the allgather.
+
+    Raises:
+        ValueError:
+            If ``nan_strategy`` is not one of ``error``, ``warn``, ``ignore`` or a float
+    """
+
+    value: Tensor
+    is_differentiable = True
+
+    def __init__(
+        self,
+        fn: str,
```

Comment:
shall we support also callable?
```suggestion
        reduction_name: str,
```